src/plugins/console/public/application/containers/editor/monaco/monaco_editor.tsx

```diff
@@ -6,13 +6,119 @@
  * Side Public License, v 1.
  */
 
-import React, { FunctionComponent, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { CodeEditor } from '@kbn/code-editor';
 import { css } from '@emotion/react';
 import { CONSOLE_LANG_ID } from '@kbn/monaco';
+import { parse } from 'query-string';
+import { debounce } from 'lodash';
+import { decompressFromEncodedURIComponent } from 'lz-string';
+import { i18n } from '@kbn/i18n';
+import { useServicesContext } from '../../../contexts';
+
+export interface EditorProps {
+  initialTextValue: string;
+}
+
+interface QueryParams {
+  load_from: string;
+}
+
+const DEFAULT_INPUT_VALUE = `# Click the Variables button, above, to create your own variables.
+GET \${exampleVariable1} // _search
+{
+  "query": {
+    "\${exampleVariable2}": {} // match_all
+  }
+}`;
+
+export const MonacoEditor = ({ initialTextValue }: EditorProps) => {
+  const {
+    services: {
```

Comment:
I'm not sure the code is using all of those services, maybe we can remove the ones that are not needed?